osquery/http/http.cpp

```diff
@@ -0,0 +1,385 @@
+/**
+Copyright (c) 2010 Daniel Schauenberg
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+This licence only applies to the files that it appears in.
+
+ * @file restclient.cpp
+ * @brief implementation of the restclient class
+ * @author Daniel Schauenberg <d@unwiredcouch.com>
+ */
+
+#include "http.h"
+
+#include <cstring>
+#include <string>
+#include <iostream>
+#include <map>
+
+namespace osquery {
+/** initialize user agent string */
+const char* RestClient::user_agent = "osquery/1.4.4";
+/** initialize authentication variable */
```

Comment:
Use https://github.com/facebook/osquery/blob/master/include/osquery/core.h#L40 for the version